### PR TITLE
Document the arrays more precisely.

### PR DESCRIPTION
**Pull request by dereuromark**
This takes care of some returned arrays.

### CODE REVIEW

**Comment by dereuromark** [comment]
> Error: Whitespace found at end of line

I cannot find the issue, using local sniff run

**Comment by othercorey** [comment]
> > Error: Whitespace found at end of line
> 
> I cannot find the issue, using local sniff run

I had the same issue.

**Comment by othercorey** [comment]
This one looks ok too.

File: src/Form/FormProtector.php [hunk 1]
```diff
@@ -380,7 +380,7 @@ protected function sortedUnlockedFields(array $formData): array
      *
      * @param string $url Form URL.
      * @param string $sessionId Session Id.
-     * @return array The token data.
+     * @return array<string, mixed> The token data.
```
Comment (ADmad) [inline src/Form/FormProtector.php:383]:
```suggestion
     * @return array<string, string> The token data.
```

File: src/Mailer/Message.php [hunk 2]
```diff
@@ -257,7 +257,7 @@ class Message implements JsonSerializable, Serializable
      *
      * Only absolute paths
      *
-     * @var array
+     * @var array<string, mixed>
```
Comment (ADmad) [inline src/Mailer/Message.php:260]:
```suggestion
     * @var array<string, array>
```

File: src/Mailer/Message.php [hunk 3]
```diff
@@ -1217,7 +1217,7 @@ public function setAttachments(array $attachments)
     /**
      * Gets attachments to the email message.
      *
-     * @return array Array of attachments.
+     * @return array<string, mixed> Array of attachments.
```
Comment (ADmad) [inline src/Mailer/Message.php:1220]:
```suggestion
     * @return array<string, array> Array of attachments.
```

File: src/Mailer/Renderer.php [hunk 4]
```diff
@@ -49,7 +49,7 @@ public function __construct()
      *
      * @param string $content The content.
      * @param array<string> $types Content types to render. Valid array values are Message::MESSAGE_HTML, Message::MESSAGE_TEXT.
-     * @return array The rendered content with "html" and/or "text" keys.
+     * @return array<string, mixed> The rendered content with "html" and/or "text" keys.
```
Comment (ADmad) [inline src/Mailer/Renderer.php:52]:
```suggestion
     * @return array<string, string> The rendered content with "html" and/or "text" keys.
```

File: src/ORM/Association/Loader/SelectLoader.php [hunk 5]
```diff
@@ -426,7 +426,7 @@ protected function _buildSubquery(Query $query): Query
      * that need to be present to ensure the correct association data is loaded.
      *
      * @param \Cake\ORM\Query $query The query to get fields from.
-     * @return array The list of fields for the subquery.
+     * @return array<string, mixed> The list of fields for the subquery.
```
Comment (ADmad) [inline src/ORM/Association/Loader/SelectLoader.php:429]:
```suggestion
     * @return array<string, array> The list of fields for the subquery.
```

**Pull request merged**